Formula/spack.rb

```diff
@@ -0,0 +1,21 @@
+class Spack < Formula
+  desc "Package manager that builds multiple versions and configurations of software"
+  homepage "https://spack.io"
+  url "https://github.com/spack/spack/releases/download/v0.16.0/spack-0.16.0.tar.gz"
+  sha256 "064b2532c70916c7684d4c7c973416ac32dd2ea15f5c392654c75258bfc8c6c2"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  head "https://github.com/spack/spack.git", branch: "develop"
+
+  bottle :unneeded
+
+  depends_on "python"
+
+  def install
+    cp_r Dir["bin", "etc", "lib", "share", "var"], prefix.to_s
```

Comment:
Looks like the tarball contains pre-built binaries. It'll need to be something built from source.

Ref. https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae